cpp/src/plasma/store.cc

```diff
@@ -454,11 +454,15 @@ void PlasmaStore::disconnect_client(int client_fd) {
   ARROW_LOG(INFO) << "Disconnecting client on fd " << client_fd;
   // If this client was using any objects, remove it from the appropriate
   // lists.
+  auto client = it->second.get();
   for (const auto& entry : store_info_.objects) {
-    if (entry.second->state == PLASMA_SEALED) {
-      remove_client_from_object_clients(entry.second.get(), it->second.get());
-    } else {
-      abort_object(entry.first);
+    auto entry_it = entry.second->clients.find(client);
+    if (entry_it != entry.second->clients.end()) {
+      if (entry.second->state == PLASMA_SEALED) {
+        remove_client_from_object_clients(entry.second.get(), client);
```

Comment:
As it is with this change, this check is a little asymetric, i.e. remove_client_from_object_clients already checks if the object is from the "right" client, but abort_object doesn't and then here we check in either case. Can you make it more symmetric and handle the checks in the same way? We could have the check in abort_object if we also pass the client and remove it from here.